cpp/test/sg/fil_test.cu

```diff
@@ -731,6 +779,16 @@ std::vector<FilTestParams> predict_sparse_inputs = {
    42, 2e-3f, tl::Operator(0), fil::leaf_algo_t::CATEGORICAL_LEAF, 3},
   {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::RAW, 0, 0, fil::algo_t::NAIVE,
    42, 2e-3f, tl::Operator(0), fil::leaf_algo_t::CATEGORICAL_LEAF, 3},
+  {20000, 50, 0.05, 2, 5000, 0.05,
```

Comment:
How long does this test take? If it takes too long, consider reducing the number of classes (300 should be enough to test the case with more classes than threads).

Reply:
`2.1s * 2 = 4.2s`. vs `1.2s * 2 = 2.4s` for similar `CATEGORICAL_LEAF`. Those are the top-3rd to top-6th tests, although they only account for several % of runtime. We can reduce the numbers and just comment that it should work on 5000?

Reply:
4.2s is fine.